src/Servers/HttpSys/test/FunctionalTests/ResponseTests.cs

```diff
@@ -210,6 +213,71 @@ public async Task Response_OnStartingThrowsAfterWrite_WriteThrowsAndStillCallsOn
             }
         }
 
+        [ConditionalFact]
+        public async Task ClientDisconnectsBeforeResponse_ResponseCanStillBeModified()
+        {
+            var readStarted = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
+            var readCompleted = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
+            using var server = Utilities.CreateHttpServer(out var address, async httpContext =>
+            {
+                var readTask = httpContext.Request.Body.ReadAsync(new byte[10]);
+                readStarted.SetResult();
+                try
+                {
+                    await readTask;
+                    readCompleted.SetException(new InvalidOperationException("The read wasn't supposed to succeed"));
+                    return;
+                }
+                catch (IOException)
+                {
+                }
+
+                // Let the Abort logic finish.
+                await Task.Yield();
```

Comment:
This doesn't seem reliable.

What about:
1. Adding a TCS
2. Using `httpContext.RequestAbort` callback to set the TCS as complete
3. Awaiting TCS here.

That ensures the following logic always happens after the server has received the client disconnection.

Reply:
With the re-ordered code in Fail I might be able to remove this entirely.

Reply:
To clarify, this Yield was only here to make the original race condition more prominent so I could verify it was fixed, the test did not require it to pass.